### PR TITLE
Update dashTable to 4.5.1 with support for async loading in R

### PR DESCRIPTION
**Pull request by rpkyle**
This PR proposes to update `dashTable` to 4.5.1, while bundling in the `df_to_list` helper function and its documentation.

This PR is related to, and required for plotly/dashR#157 to function properly.

### CODE REVIEW

File: dash_table/async~table.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
...f(r)>=0||Object.prototype.hasOwnProperty.call(e,r)&&(n[r]=e[r]);return n},Cr=function(e,t){if(!e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!t||"object"!=typeof t&&"function"!=typeof t?e:t},jr=function(e){function t(e){Or(this,t);var n=Cr(this,(t.__proto__||Object.getPrototypeOf(t)).call(this,e));return n.handleMouseDown=n.handleMouseDown.bind(n),n.handleMouseEnter=n.handleMouseEnter.bind(n),n.handleMouseMove=n.handleMouseMove.bind(n),n.handleTouchStart=n.handleTouchStart.bind(n),n.handleTouchEnd=n.handleTouchEnd.bind(n),n.handleTouchMove=n.handleTouchMove.bind(n),n.onFocus=n.onFocus.bind(n),n}return Er(t,e),kr(t,[{key:"handleMouseDown",value:function(e){e.preventDefault(),e.stopPropagation(),this.props.onSelect(this.props.option,e)}},{key:"handleMouseEnter",value:function(e){this.onFocus(e)}},{key:"handleMouseMove",value:function(e){this.onFocus(e)}},{key:"handleTouchEnd",value:function(e){this.dragging||this.handleMouseDown(e)}},{key:"handleTouchMove",value:function(){this.dragging=!0}},{key:"handleTouchStart",value:function(){this.dragging=!1}},{key:"onFocus",value:function(e){this.props.isFocused||this.props.onFocus(this.props.op
```
Comment (Marc-Andre-Rivet) [inline dash_table/async~table.js]: These diffs are annoying but they boil down to different ordering for resource inclusion.. as long as all bundles are updated, there's no problem.

File: dash-info.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-pkg_help_description: >
-    Dash DataTable is an interactive table component designed for
-    designed for viewing, editing, and exploring large datasets.
-    DataTable is rendered with standard, semantic HTML <table/> markup,
-    which makes it accessible, responsive, and easy to style. This
-    component was written from scratch in React.js specifically for the
-    Dash community. Its API was designed to be ergonomic and its behaviour
-    is completely customizable through its properties.
-pkg_help_title: >
+pkg_help_description: >-
```
Comment (Marc-Andre-Rivet) [inline dash-info.yaml]: Is this file generated?

File: R/df_to_list.R [hunk 3]
```diff
@@ -0,0 +1,8 @@
+df_to_list <- function(df) {
```
Comment (Marc-Andre-Rivet) [inline R/df_to_list.R]: Is this file generated?

File: dash-info.yaml (hunk 2 above)
Comment (rpkyle) [reply dash-info.yaml]: @Marc-Andre-Rivet The `dash-info.yaml` file is "handwritten", so it's something we'll carry forward but won't update using the package generator.

File: R/df_to_list.R (hunk 3 above)
Comment (rpkyle) [reply R/df_to_list.R]: @Marc-Andre-Rivet This function was also user-authored (by me), to emulate the syntax on the Python side using `df_to_dict`. R users reported that it made using `dashDataTable` easier for them, so it's persisted. It turns out that it's useful for `dashBio` as well.

File: dash-info.yaml (hunk 2 above)
Comment (Marc-Andre-Rivet) [reply dash-info.yaml]: Ok, so what I see is that this code (and other similar files) will need to be merged back into `dev` once merged into master or be committed into `dev` first, merged into `master` and this branch updated.

Comment (rpkyle) [reply dash-info.yaml]: Right, fortunately the only non-build artifacts are these YAML files (I believe), so I'll cherry-pick those into `dev` and remove from this PR before we merge.

File: R/df_to_list.R (hunk 3 above)
Comment (rpkyle) [reply R/df_to_list.R]: I think this one is OK to leave in `master`, along with the rest of the R package items (vignettes, Rd files, etc).

File: dash-info.yaml (hunk 2 above)
Comment (rpkyle) [reply dash-info.yaml]: fixed via 

https://github.com/plotly/dash-table/pull/669
https://github.com/plotly/dash-html-components/pull/147
https://github.com/plotly/dash-core-components/pull/722

File: DESCRIPTION [hunk 4]
```diff
@@ -1,11 +1,10 @@
 Package: dashTable
 Title: Core Interactive Table Component for Dash
 Version: 4.5.1
-Authors @R: as.person(c(Chris Parmer <chris@plot.ly>))
-Description:  Dash DataTable is an interactive table component designed for designed for viewing, editing, and exploring large datasets. DataTable is rendered with standard, semantic HTML <table/> markup, which makes it accessible, responsive, and easy to style. This component was written from scratch in React.js specifically for the Dash community. Its API was designed to be ergonomic and its behavior is completely customizable through its properties.
+Description: Dash DataTable is an interactive table component designed for designed for viewing, editing, and exploring large datasets. DataTable is rendered with standard, semantic HTML <table/> markup, which makes it accessible, responsive, and easy to style. This component was written from scratch in React.js specifically for the Dash community. Its API was designed to be ergonomic and its behaviour is completely customizable through its properties.
```
Comment (Marc-Andre-Rivet) [inline DESCRIPTION]: Shouldn't start with `Dash DataTable` for mysterious reasons

File: man/dashTable-package.Rd [hunk 5]
```diff
@@ -2,11 +2,9 @@
 \docType{package}
 \name{dashTable-package}
 \alias{dashTable}
-\title{Core Interactive Table Component for Dash
-}
+\title{Core Interactive Table Component for Dash}
 \description{
 Dash DataTable is an interactive table component designed for designed for viewing, editing, and exploring large datasets. DataTable is rendered with standard, semantic HTML <table/> markup, which makes it accessible, responsive, and easy to style. This component was written from scratch in React.js specifically for the Dash community. Its API was designed to be ergonomic and its behaviour is completely customizable through its properties.
```
Comment (Marc-Andre-Rivet) [inline man/dashTable-package.Rd]: Shouldn't start with `Dash DataTable` for mysterious reasons

File: DESCRIPTION (hunk 4 above)
Comment (rpkyle) [reply DESCRIPTION]:
```suggestion
Description: An interactive table component designed for designed for viewing, editing, and exploring large datasets, DataTable is rendered with standard, semantic HTML <table/> markup, which makes it accessible, responsive, and easy to style. This component was written from scratch in React.js specifically for the Dash community. Its API was designed to be ergonomic and its behaviour is completely customizable through its properties.
```

**Pull request merged**